File: src/Sevostyanov D/SsauAcceptanceTests/src/test/java/pages/BasePage.java
package pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.remote.RemoteWebDriver;

import java.util.NoSuchElementException;
import java.util.concurrent.TimeUnit;

public class BasePage {

    protected RemoteWebDriver driver;

    public BasePage(RemoteWebDriver driver) {
        this.driver = driver;

        driver.manage().timeouts().implicitlyWait(10, TimeUnit.SECONDS);
        driver.manage().window().maximize();
    }

    public WebElement findElementByCss(String css) {
        WebElement element;

        try {
            element = driver.findElement(By.cssSelector(css));
        } catch (NoSuchElementException ex) {
            throw new NoSuchElementException("CSS selector = \"" + css + "\"\n");
        }

        return element;
    }

    public void openPage(String pagePath) {
        driver.navigate().to(pagePath);
    }
}

File: src/Nesterova/SsauScheduleTest/SsauScheduleTest/PageObjects/BasePage.cs
using OpenQA.Selenium;
using System;
using System.Collections.Generic;
using System.Text;

namespace SsauScheduleTest
{
    public class BasePage
    {
        protected IWebDriver Driver { get; private set; }
        public BasePage(IWebDriver driver, double timeout=10.0)
        {
            Driver = driver;
            Driver.Manage().Timeouts().ImplicitWait = TimeSpan.FromSeconds(timeout);
            Driver.Manage().Window.Maximize();
        }
        public IWebElement FindElementByCss(string css)
        {
            IWebElement element;
            try
            {
                element = Driver.FindElement(By.CssSelector(css));
            }
            catch (NoSuchElementException ex)
            {
                throw new NoSuchElementException(
                    string.Format("CSS selector = \"{0}\"\n", css),
                    ex);
            }
            return element;
        }
        public void OpenPage(string pagePath)
        {
            Driver.Navigate().GoToUrl(pagePath);
        }
    }
}
